[Lim Gang Le] iP

limgl1998 opened the pull request.

limgl1998 committed:
Level 0. Greet

Implement an initial skeletal version of the Duke that simply greets the user and exits.

limgl1998 committed:
Level 1. Greet, Echo, Exit

Implement a skeletal version of Duke that starts by greeting the user, simply echos commands entered by the user, and exits when the user types bye.

limgl1998 committed:
Level 2. Add, List

Add the ability to store whatever text entered by the user and display them back to the user when requested.

limgl1998 committed:
Level 3. Mark as Done

Add the ability to mark tasks as done.

limgl1998 committed:
A-CodingStandard

Edited code to follow the Coding Standard

limgl1998 committed:
Level 4. ToDos, Events, Deadlines

Add support for tracking three types of tasks:

ToDos: tasks without any date/time attached to it e.g., visit new theme park
Deadlines: tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm
Events: tasks that start at a specific time and ends at a specific time e.g., team project meeting on 2/10/2019 2-4pm

limgl1998 committed:
Level 4.1

Introduced enum to replace some "magic words" such as todo and event
Added some methods to handle invalid inputs

pinfang commented on src/main/java/CommandType.java:
```diff
@@ -0,0 +1,3 @@
+public enum CommandType {
+    todo, event, deadline, done
```
Shouldn't these be all capital letters? (TODO, EVENT, ... )

pinfang commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,19 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Scanner in = new Scanner(System.in);
```
Is "in" a good naming?

pinfang commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task{
```
Should there be a space after Task ?
I noticed the same issue in several other places too.

limgl1998 committed:
A-CodeQuality

Improve Code Quality. Refactored some duplicate codes into methods and shifted some general methods in to another class called GeneralMethods

limgl1998 committed:
A-TextUiTesting: Automated Text UI Testing

Added ability to test for some of the possible invalid inputs
Fixed a minor bug for handling input "list"

e0426051 commented on src/main/java/HandleInput.java:
```diff
@@ -0,0 +1,32 @@
+
+    public static void handlesInput(String input){
+
+        Message.printDashedLine();
```
Good readable code, no deep nesting/arrowhead style

e0426051 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,17 @@
+import java.util.Scanner;
```
In general the expressions are relatively simple. Also no magic numbers

e0426051 replied:
The code is also largely KISS compatible, easy for readers
SLAP is also implemented appropriately

pinfang commented on src/main/java/HandleInput.java:
```diff
@@ -0,0 +1,32 @@
+public class HandleInput {
```
You have separated a class to handle the commands. I like it.

pinfang commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,13 @@
+public class Event extends Task{
```
This is good.

pinfang commented on src/main/java/GeneralMethods.java:
```diff
@@ -0,0 +1,23 @@
+public class GeneralMethods {
```
I like how clean this bit of code is.

pinfang commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,84 @@
+public class TaskList {
+    private static final int NUMBER_OF_PARTS = 2;
+
+    private Task[] list;
+    private static int numberOfTasks;
```
The naming are good and easy to understand.

pinfang commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,84 @@
+    public void markAsDone(String command) {
+        command = GeneralMethods.removeCommandFromInput(command , CommandType.done);
+        //Error handling for input "done" without task number
+        if(command.isEmpty()|| !GeneralMethods.isNumeric(command)) {
```
Should there a spacing after the command.isEmpty()?

pinfang commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,11 @@
+public class ToDo extends Task{
```
This is good.

pinfang reviewed (commented):
Overall, I think your code is clean and easy to understand. You have categorized your commands and messages in different classes which is great.

limgl1998 committed:
Fixed the indentation errors

e0426051 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,22 @@
+        return ("[" + (isDone ? "\u2713" : "\u2718") + "] "); //return tick or X symbols
+    }
+
+    public String getStatusAndDescription() {
```
Good SLAP implementation to de-clutter the main function.

limgl1998 committed:
Level 5. Handle Errors

Teach Duke to deal with errors such as incorrect inputs entered by the user.
Edited some of the old error messages and added some new ones.

limgl1998 committed:
Merge remote-tracking branch 'origin/heads/branch-Level-5'

limgl1998 committed:
Fixed indentation errors

limgl1998 committed:
Refactored the files into packages

limgl1998 committed:
Updated the src directories in batch file

limgl1998 committed:
Add Increment: A-Packages

limgl1998 committed:
Merge branch 'branch-A-Packages'

limgl1998 committed:
Included use of exceptions in the error handling

limgl1998 committed:
Replaced a magic number and included a new exception

limgl1998 committed:
Updated the batch file and edited some error handling

Batch file directory was updated. Error handling for addTask was updated.

limgl1998 committed:
A-Collections

Changed the implementation of the task list from task[] to Arraylist

limgl1998 committed:
Level 6. Delete

Add support for deleting tasks from the list.

limgl1998 committed:
Level 7. Save

Save the tasks in the hard disk automatically whenever the task list changes. Load the data from the hard disk when Duke starts up.

limgl1998 committed:
Merge branch 'branch-Level-6'

limgl1998 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/duke/Command/HandleInput.java
#	src/main/java/duke/Tasks/TaskList.java

limgl1998 committed:
Added new error handling for printing of empty task list

limgl1998 committed:
Fixed some bugs in the code

Resolved the problem when the code displays the messages of task being when first started
Resolved the numbering of tasks

limgl1998 committed:
Fixed some naming issues

limgl1998 committed:
Renamed several classes

limgl1998 committed:
Add Increment: A-MoreOOP

Refactor the code to extract out closely related code as classes.

limgl1998 committed:
Changed to access storage by instance

limgl1998 committed:
Updated Duke.java to become more OOP

limgl1998 committed:
Level 9. Find

Give users a way to find a task by searching for a keyword.

limgl1998 committed:
A-JavaDoc

Add JavaDoc comments

limgl1998 committed:
Merge pull request #1 from limgl1998/heads/A-JavaDoc

A-JavaDoc

limgl1998 committed:
Merge remote-tracking branch 'origin/master' into branch-Level-9

limgl1998 committed:
Merge pull request #2 from limgl1998/branch-Level-9

Branch level 9

limgl1998 committed:
Even more OOP

Added more comments
Changed enum to all capital letter
Refactored some commonly used strings to improve readability

Bencotti commented on src/main/java/duke/Command/Parser.java:
```diff
@@ -0,0 +1,71 @@
+     * @param printMessage
+     * @return is input BYE
+     */
+    public static boolean handlesInput(String input, boolean printMessage) {
```
Following the coding standards guide, the variable `printMessage` could be renamed to sound more like a boolean instead of a statement

Bencotti commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,47 @@
+        }
+    }
+
+    private void start() {
+        ui.printStart();
+    }
+
+    private void end() {
+        ui.printExit();
+    }
+
+    public void run() {
+        start();
+        runUntilByeCommand();
+        end();
+    }
+
+    public static void main(String[] args) {
+        new Duke().run();
+    }
```
Great job applying SLAP principle! 👍

Bencotti commented on src/main/java/duke/FileIO/Storage.java:
```diff
@@ -0,0 +1,120 @@
+        if (doesFileNotExist()) {
+            return;
+        }
+        File data = new File("data/data.txt");
```
You could consider saving the file path as a constant variable since it remains static, it would make it easier to find and change if needed in the future

limgl1998 replied:
Okay sure, will update in a later version

Bencotti commented on src/main/java/duke/GeneralMethods.java:
```diff
@@ -0,0 +1,41 @@
+     *
+     * @param input
+     * @param command
+     * @return
```
Should explain to viewers what is being returned

Bencotti commented on src/main/java/duke/Tasks/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+ */
+
+public class Deadline extends Task {
+    protected String by;
```
Perhaps consider using a variable name that better explains the details of the data it holds? This would make it easier to understand the code at a glance and avoid possible confusion

Bencotti commented on src/main/java/duke/Tasks/TaskList.java:
```diff
@@ -0,0 +1,244 @@
+ */
+
+public class TaskList {
+    private static final int NUMBER_OF_PARTS = 2;
```
Good use of constant variable to explain what this magic number means. Even though it might seem obvious why you'd use certain numbers in code, after some time you may forget the rational. This makes it easy to read and understand what the logic does! 👍

Bencotti reviewed (commented):
Good job trying to keep to the coding standards and applying the code readability techniques taught in class! 👍 Perhaps something to look into would be see how else to improve readability, for example having more value-added variable names or using less magic numbers. Also, a number of your Javadoc comments were left uncomplete. You're almost there, just a bit more before the final submission! 👍

limgl1998 committed:
Added Help command

Help command shows all the availiable commands

limgl1998 committed:
Updated the read from file method

Read from file method is now non static

limgl1998 committed:
Even even more OOP

updated the magic numbers/ strings to constants
filled in the params for methods whose params are empty

limgl1998 committed:
Updated the README, Find method updated

README updated to contain the user guide
Removed the extra lines printed

limgl1998 committed:
Attempted fix for formating issues in the UG

limgl1998 committed:
Fixed formating issues in UG

limgl1998 committed:
Set theme jekyll-theme-minimal

limgl1998 committed:
Updated the UG

Changed the descriptions of some features
Added function of saving upon update

limgl1998 committed:
Merge branch 'master' of https://github.com/limgl1998/ip

limgl1998 committed:
Minor changes to the UG

Removed a full stop

limgl1998 committed:
Fixed some formating issues

limgl1998 committed:
Updated the description for event
